src/Microsoft.ServiceFabric.Actors/Diagnostics/MetricDiagnostics.cs

```diff
@@ -0,0 +1,125 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Fabric;
+using Microsoft.ServiceFabric.Diagnostics;
+using Microsoft.ServiceFabric.Diagnostics.Metrics;
+
+namespace Microsoft.ServiceFabric.Actors.Diagnostics
+{
+    internal class MetricDiagnostics : IDiagnostics
+    {
+        readonly IClock clock;
+
+        readonly IMeter<long> actorLockContention;
+        readonly IMeter<TimeSpan> acquireLockDuration;
+        readonly IMeter<TimeSpan> releaseLockDuration;
+        readonly IMeter1D<long> methodExceptionCount;
+        readonly IMeter1D<TimeSpan> methodExecutionDuration;
+        readonly IMeter<TimeSpan> onActivateAsyncDuration;
+        readonly IMeter<TimeSpan> requestProcessingDuration;
+        readonly IMeter<TimeSpan> loadStateDuration;
+        readonly IMeter<TimeSpan> saveStateDuration;
+
+        public MetricDiagnostics(IMeterProvider<long> longMeterProvider, IMeterProvider<TimeSpan> timeSpanProvider, IClock clock)
+        {
+            _ = longMeterProvider ?? throw new ArgumentNullException(nameof(longMeterProvider));
+            _ = timeSpanProvider ?? throw new ArgumentNullException(nameof(timeSpanProvider));
+            this.clock = clock ?? throw new ArgumentNullException(nameof(clock));
+
+            this.actorLockContention = longMeterProvider.CreateMeter("Actor", "ActorLockContention");
+            this.acquireLockDuration = timeSpanProvider.CreateMeter("Actor", "AcquireLockDuration");
+            this.releaseLockDuration = timeSpanProvider.CreateMeter("Actor", "ReleaseLockDuration");
+            this.methodExceptionCount = longMeterProvider.CreateMeter("Actor", "MethodExceptionCount", "MethodId");
+            this.methodExecutionDuration = timeSpanProvider.CreateMeter("Actor", "MethodExecutionDuration", "MethodId");
```

Comment:
Are MethodIds stable between processes, nodes, app versions, etc? At the minimum we should be able to convert them back to the actor names long time after the app was built and deployed. I think it would be best to use method name, possibly method signature as the dimension value, like the EventSourceDiagnostics does.